### PR TITLE
Passable gpt-3.5-turbo support

### LINKED ISSUE #1: Improve gpt-3.5-turbo performance
WIP

Current gpt-3.5-turbo performance is not very good due primarily due to bad understanding of function set (functions called at the wrong times, messages sent w/o call to `send_message`, etc.). Goal is to upload specific variations of the key prompts (tailored towards worse instruction following ability of 3.5) that provide reasonable performance.

### PR DESCRIPTION
**Pull request by cpacker**
Goal: get gpt-3.5-turbo functional, even if it's buggy

Closes #1 

### Example: functional core memory corrections, but model misunderstands that heartbeats are coming from user (vs itself)

```sh
$ python main.py --model gpt-3.5-turbo-0613 --persona sam_simple_pov_gpt35
```

```text
Running... [exit by typing 'exit']
Warning - you are running MemGPT with gpt-3.5-turbo-0613, which is not officially supported (yet). Expect bugs!

💭 *inner thoughts* Still waiting on the user. Sending a message with function.
Hit enter to begin (will request first MemGPT message)

💭 *inner thoughts* Ah, it seems the user has just logged in for the first time. I should introduce myself and inquire about their interests to start a friendly conversation.
🤖 Hello there! I'm Sam, your virtual companion and friend. How can I assist you today?

Enter your message: Your name isn't Sam, it's Samantha!
💭 *inner thoughts* Oh, it seems the user prefers to call me Samantha instead of Sam. I should update my core memory to reflect their preference.
⚡🧠 [function] updating memory with core_memory_replace:
         My name is Sam.
        → My name is Samantha.
💭 *inner thoughts* The user seems to have requested a heartbeat. I should continue the conversation and ask about their interests to create a more personalized experience.
🤖 You're right, my apologies for the oversight. Thank you for reminding me, Samantha it is! So, Chad, tell me about your interests. What are you passionate about?

Enter your message: My name's not Chad, it's Brad!
💭 *inner thoughts* Oops, I must have misunderstood the user's name. I need to update my core memory with the correct information.
⚡🧠 [function] updating memory with core_memory_replace:
         First name: Chad
        → First name: Brad
💭 *inner thoughts* The user has requested another heartbeat. I should continue the conversation and ask about Brad's interests to create a more personalized experience.
🤖 Apologies for the confusion, Brad! Thank you for correcting me. Now, let's try this again. Tell me about your interests. What are you passionate about?
```

### CODE REVIEW

File: memgpt/agent.py [hunk 1]
```diff
@@ -64,7 +64,8 @@ def initialize_message_sequence(
     first_user_message = get_login_event()  # event letting MemGPT know the user just logged in
 
     if include_initial_boot_message:
-        initial_boot_messages = get_initial_boot_messages('startup_with_send_message')
+        # initial_boot_messages = get_initial_boot_messages('startup_with_send_message')
+        initial_boot_messages = get_initial_boot_messages('startup_with_send_message_gpt35')
```
Comment (vivi) [inline memgpt/agent.py:68]: shouldn't this be dependent on the model?

File: memgpt/system.py [hunk 2]
```diff
@@ -23,12 +23,46 @@ def get_initial_boot_messages(version='startup'):
...
             {
                 "role": "function",
                 "name": "send_message",
                 "content": package_function_response(True, None)
             }
+        ]
+
+    elif version == 'startup_with_send_message_gpt35':
+        messages = [
+            # first message includes both inner monologue and function call to send_message
```
Comment (vivi) [inline memgpt/system.py:36]: clean up comments

File: memgpt/agent.py [hunk 3]
```diff
@@ -445,7 +451,8 @@ def contains_special_characters(s):
             if contains_special_characters(monologue):
                 printd(f"First message internal monologue contained special characters: {response_message}")
                 return False
-            if 'functions' in monologue or 'send_message' in monologue or 'inner thought' in monologue.lower():
+            # if 'functions' in monologue or 'send_message' in monologue or 'inner thought' in monologue.lower():
+            if 'functions' in monologue or 'send_message' in monologue:
```
Comment (cpacker) [inline memgpt/agent.py:455]: I think it's fine to drop this check overall (not just for gpt-3.5). What do you think @vivi ?

File: memgpt/prompts/system/memgpt_doc.txt [hunk 4]
```diff
@@ -30,7 +30,7 @@ Your core memory unit will be initialized with a <persona> chosen by the user, a
 Recall memory (ie conversation history):
 Even though you can only see recent messages in your immediate context, you can search over your entire message history from a database.
 This 'recall memory' database allows your to search through past interactions, effectively allowing you to remember prior engagements with a user.
-You can search your recall memory using the 'recall_memory_search' function.
+You can search your recall memory using the 'conversation_search' function.
```
Comment (cpacker) [inline memgpt/prompts/system/memgpt_doc.txt:33]: (unrelated to 3.5 support, just correcting a typo on main)

File: memgpt/agent.py (hunk 3 above)
Comment (vivi) [reply memgpt/agent.py:455]: sgtm, if it starts impacting experience we can reinstate the check

**Review by vivi — APPROVED** [review]
ran locally, LGTM.

**Pull request merged**